### PR TITLE
School Info Confirmation Dialog UI Test

### PR DESCRIPTION
**Pull request by nkiruka**
Ui test to test when a user sees the school confirmation dialog and school info interstitial

The user scenarios tested:
- User enters partial school info
- One week later, teacher sees school info interstitial and completes the form 
- One week later, teacher does not receive prompt to complete school info 
- One year later, teacher sees  the confirmation dialog

### Ui-test
ui_test_school_info

### CODE REVIEW

File: dashboard/test/ui/features/school_info_confirmation_dialog.feature [hunk 1]
```diff
@@ -0,0 +1,8 @@
+Feature: School Info Confirmation Dialog
+
+Scenario: One week later, user receives prompt to complete school info interstitial
```
Comment (Hamms) [inline dashboard/test/ui/features/school_info_confirmation_dialog.feature]:
```suggestion
Scenario: One week after account creation, user receives prompt to complete school info interstitial
```

File: dashboard/test/ui/features/step_definitions/steps.rb [hunk 2]
```diff
@@ -1711,3 +1711,10 @@ def get_section_id_from_table(row_index)
   body_text = @browser.execute_script('return document.body && document.body.textContent;').to_s
   expect(body_text.include?(text)).to eq(negation.nil?)
 end
+
+And (/^I update user "([^"]*)" account creation date$/) do |name|
```
Comment (Hamms) [inline dashboard/test/ui/features/step_definitions/steps.rb]: I don't love the name here; "update account creation date" is fairly generic, and this is doing something not only specific, but rather strange; updating the user's creation date to a week ago.

I'd prefer to see name that's more specific, and also a comment explaining why this is a useful step to have

File: dashboard/test/ui/features/school_info_confirmation_dialog.feature [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+# This test checks three relevant states of the user in the school info confirmation
+# dialog flow:
+# 1. The initial flow is when a user creates a new account and does not fill out
+# all the fields in the school info interstitial.
+# 2. The user receives prompt to complete school info interstitial 7 days after
+# account creation and completes school info.
+# 3. A week later, the user should not receive a prompt to complete school info
+# 4. A year later after the user has completed school info, the user sees prompt to
+# confirm or update current school info.
```
Comment (islemaster) [inline dashboard/test/ui/features/school_info_confirmation_dialog.feature:12]: Great overview!

File: dashboard/test/ui/features/step_definitions/steps.rb [hunk 4]
```diff
@@ -1711,3 +1711,25 @@ def get_section_id_from_table(row_index)
...
+end
+
+And(/^I update user "([^"]*)" last_seen_school_info_interstitial$/) do |name|
+  require_rails_env
+  user = User.find_by_email_or_hashed_email(@users[name][:email])
+  user.last_seen_school_info_interstitial = 8.days.ago
+  user.save!
+end
+
+And(/^I update user "([^"]*)" last_confirmation_date and last_seen_school_info_interstitial$/) do |name|
```
Comment (islemaster) [inline dashboard/test/ui/features/step_definitions/steps.rb]: These steps don't tell the whole story of what the step is doing.  For example, it's a little surprising that
```
I update user "Sam" last_seen_school_info_interstitial
```
effectively ages the account eight days, but
```
I update user "Sam" last_confirmation_date and last_seen_school_info_interstitial
```
ages it a whole year.

Let's update the names to mention something about the time passing.

Totally optional: I bet we could come up with clearer names, deduplicate some code, and be a little more complete in our simulation of time passing with something like this:

```rb
And /^Eight days pass for user "([^"]*)"$/ do |name|
  pass_time_for_user name, 8.days
end

And /^One year passes for user "([%"]*)"$/ do |name|
  pass_time_for_user name, 1.year
end

def pass_time_for_user(user_name, amount_of_time)
  require_rails_env
  user = User.find_by_email_or_hashed_email(@users[name][:email])
  user.created_at -= amount_of_time
  user.last_seen_school_info_interstitial -= amount_of_time if user.last_seen_school_info_interstitial
  user.save!
  user.user_school_infos.each do |info|
    info.last_confirmation_date -= amount_of_time
    info.save!
  end
end
```

**Review by islemaster — APPROVED** [review]
Great testing!  Thanks for putting together thorough coverage for this feature.

**Pull request merged**